src/overlays/actors/ovl_Dm_Char09/z_dm_char09.c

```diff
@@ -32,28 +30,197 @@ const ActorInit Dm_Char09_InitVars = {
+        if (phi_a1) {
+            func_80AB1FA0(this, this->unk_21C);
+        }
+    }
+    Math_SmoothStepToS(&thisx->world.rot.y, Math_Vec3f_Yaw(&thisx->world.pos, &sp40) + 0x7FFF, 1, 0x7D0, 0);
+    thisx->shape.rot.y = thisx->world.rot.y;
+}
+
+void DmChar09_DoNothing(DmChar09* this, PlayState* play) {
+}
+
+void func_80AB2268(DmChar09* this, PlayState* play) {
+    Path* path;
+    s32 pad;
+    s32 i;
+    s32 actionIndex;
+    s32 action = 0;
+    s32 pathnum;
+    u8 csIndex = 0;
+
+    if (!DMCHAR09_GET_F(&this->actor)) {
+        if (play->csCtx.currentCsIndex == 1) {
+            csIndex = 1;
+        }
+    } else if (play->csCtx.currentCsIndex == 0) {
+        csIndex = 1;
+    }
+
+    if (Cutscene_CheckActorAction(play, 0x1F7) && (csIndex)) {
+        actionIndex = Cutscene_GetActorActionIndex(play, 0x1F7);
+        if (this->unk_22F != play->csCtx.actorActions[actionIndex]->action) {
+            this->unk_22F = play->csCtx.actorActions[actionIndex]->action;
+            switch (play->csCtx.actorActions[actionIndex]->action) {
+                case 2:
+                    action = 0;
+                    break;
+                case 3:
+                    action = 1;
+                    break;
+                case 4:
+                    action = 2;
+                    break;
+                case 5:
+                    action = 3;
+                    break;
+            }
+
+            if (play->csCtx.actorActions[actionIndex]->action >= 2) {
+                pathnum = DMCHAR09_GET_PATH(&this->actor);
+                path = &play->setupPathList[pathnum];
+
+                for (i = 0; i < action; i++) {
+                    pathnum = path->unk1;
+                    path = &play->setupPathList[pathnum];
+                }
+
+                this->unk_224 = Lib_SegmentedToVirtual(path->points);
+                this->unk_214 = path->count;
+                this->unk_21C = 0;
+                this->unk_218 = path->count - 1;
+                this->unk_220 = 1;
+                this->unk_22E = 1;
+
+                this->speed = (u16)play->csCtx.actorActions[actionIndex]->rot.z * 0.00390625f;
+                this->actionFunc = func_80AB1FDC;
+            } else {
+                this->unk_22E = 0;
+                this->actionFunc = DmChar09_DoNothing;
+            }
+        }
+    } else {
+        this->unk_22F = 0x63;
+    }
+}
+
+void func_80AB24BC(DmChar09* this, PlayState* play) {
+    this->unk_228 += 0xBB8;
+    this->unk_22A += 0x1388;
+    this->unk_204 = Math_SinS(this->unk_228) * 150.0f;
+    this->unk_208 = Math_SinS(this->unk_22A) * 150.0f;
+    this->unk_20C = Math_SinS(this->unk_228) * 150.0f;
+}
+
+void DmChar09_Update(Actor* thisx, PlayState* play) {
+    DmChar09* this = THIS;
+
+    SkelAnime_Update(&this->skelAnime);
+    this->actionFunc(this, play);
+    func_80AB2268(this, play);
+    func_80AB24BC(this, play);
+    if ((play->csCtx.state != 0) && (this->unk_22E) && DMCHAR09_GET_100(thisx)) {
```

Comment:
```suggestion
    if ((play->csCtx.state != 0) && this->unk_22E && DMCHAR09_GET_100(thisx)) {
```
Also could you change the assignments to `unk_22E` to use `true/false` instead of `1/0`